workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RenderWorkflow.kt

```diff
@@ -170,15 +171,15 @@ public fun <PropsT, OutputT, RenderingT> renderWorkflowIn(
   }
 
   scope.launch {
-    while (isActive) {
+    outer@ while (isActive) {
```

Comment:
If this code is getting complex enough that we're starting to need labels, it's probably a sign that it should be factored down a bit.

Reply:
I thought the labels were so nice and clean :D. but yes, perhaps refactoring some functions might be better?

Reply:
i don't want to do this right now as it should be done very carefully watching for its own behavior impacts because the methods we refactor out will need to handle the different cases of killing the scope vs. restarting outer loop vs continuing inner loop.